ooniprobe/View/OONIRun/OoniRunViewController.m

```diff
@@ -49,27 +50,35 @@ -(void)handleUrlScheme{
      NSLog(@"dict: %@", dict);
     
     NSString *action;
-    if ([[url host] isEqualToString:@"run.ooni.io"])
+    NSUserDefaults *usrInfo = [[NSUserDefaults alloc] initWithSuiteName:@"group.org.openobservatory.ooniprobe"];
+
+    if ([[url host] isEqualToString:@"run.ooni.io"]) {
         action = [[url path] substringFromIndex:1];
-    else
+    } else if ([[[usrInfo dictionaryRepresentation] allKeys] containsObject:@"incomingURL"] && dict[@"tn"] == nil) {
```

Comment:
What is the purpose of checking for `dict[@"th"] == nil`?

Reply:
Once the URL is shared, the next OONI Run launch still contains `incomingURL` in `usrInfo`.
Which means the only way to be sure its share action is also to check that `dict[@"tn"]` which is available only when the app is launched through OONI Run Link is absent.